test/Core.Test/Services/GroupServiceTests.cs

```diff
@@ -30,12 +33,116 @@ public GroupServiceTests()
             );
         }
 
-        // Remove this test when we add actual tests. It only proves that
-        // we've properly constructed the system under test.
         [Fact]
-        public void ServiceExists()
+        public async Task SaveAsync_DefaultGroupId_CreatesGroupInRepository()
         {
-            Assert.NotNull(_sut);
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+                UseGroups = true,
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await _sut.SaveAsync(group);
+
+            await _groupRepository.Received().CreateAsync(group);
+        }
+
+        [Fact]
+        public async Task SaveAsync_NonExistingOrganizationId_ThrowsBadRequest()
+        {
+            var group = new Models.Table.Group
+            {
+                Id = Guid.NewGuid(),
+            };
+
+            await Assert.ThrowsAsync<BadRequestException>(() => _sut.SaveAsync(group));
+        }
+
+        [Fact]
+        public async Task SaveAsync_OrganizationDoesNotUseGroups_ThrowsBadRequest()
+        {
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await Assert.ThrowsAsync<BadRequestException>(() => _sut.SaveAsync(group));
```

Comment:
Can you please validate some portion of the message to ensure you're catching the right `BadRequestException`?